src/vos/vos_io.c

```diff
@@ -533,9 +543,26 @@ vos_ioc_reserve_init(struct vos_io_context *ioc, struct dtx_handle *dth)
 	for (i = 0; i < ioc->ic_iod_nr; i++) {
 		daos_iod_t *iod = &ioc->ic_iods[i];
 
+		if (iod->iod_type == DAOS_IOD_SINGLE) {
+			gang_nr = vos_irec_gang_nr(ioc->ic_cont->vc_pool, iod->iod_size);
+			if (gang_nr > UINT8_MAX) {
+				D_ERROR("Too large SV:"DF_U64", gang_nr:%u\n",
+					iod->iod_size, gang_nr);
+				return -DER_REC2BIG;
```

Comment:
So we cannot support SV if its size is larger than 8MB * 2^8 = 2GB ? That seems not reasonable. Can we force to disable gang_addr (and try to allocate continuous large space) for the SV under such case?

Reply:
I think this number (2GB) is far more larger than the restrictions of current implementation. Without this gang SV, we can't even support 200MB sized SV.

I think we shouldn't try to allocate 2GB and hope it won't fail, furthermore, there is only by default 1GB DMA buffer per-xstream, it can't support DMA transfer for a 2GB SV.

Reply:
I am not sure how the real use case will be, such as for HDF5, hopes 2GB is large enough. Maybe we can add something in our document to describe related restriction.

Reply:
Yes, I agree that we'd document all the DAOS API restrictions well, but I think that's not a trivial work and should be done in a separate PR.